### PR TITLE
dts/arm/st: Add USB support for stm32l072/73

### PR DESCRIPTION
**Pull request by ydamigos**
Add USB support for stm32l072 and stm32l073 SoCs

Signed-off-by: Yannis Damigos <giannis.damigos@gmail.com>

### CODE REVIEW

File: arch/arm/soc/st_stm32/stm32l0/dts.fixup [hunk 1]
```diff
@@ -42,4 +42,13 @@
 #define CONFIG_SPI_2_NAME                       ST_STM32_SPI_40003800_LABEL
 #define CONFIG_SPI_2_IRQ                        ST_STM32_SPI_40003800_IRQ_0
 
+
```
Comment (tagunil) [inline arch/arm/soc/st_stm32/stm32l0/dts.fixup]: Do we need this empty line here?

**Comment by tagunil** [comment]
@ydamigos It waits forever for HSI48 readiness. Will investigate.

**Comment by tagunil** [comment]
Well, it works after enabling HSI48 buffer in SYSCFG:
```
--- a/drivers/usb/device/usb_dc_stm32.c
+++ b/drivers/usb/device/usb_dc_stm32.c
@@ -219,6 +219,10 @@ static int usb_dc_stm32_clock_enable(void)
         * STM32F030x4/x6/x8/xC and STM32F070x6/xB.
         */
 #if defined(RCC_HSI48_SUPPORT)
+#ifdef CONFIG_SOC_SERIES_STM32L0X
+       LL_SYSCFG_VREFINT_EnableHSI48();
+#endif /* CONFIG_SOC_SERIES_STM32L0X */
+
        LL_RCC_HSI48_Enable();
        while (!LL_RCC_HSI48_IsReady()) {
                /* Wait for HSI48 to become ready */
```
However, there is a problem: where should SYSCFG itself be enabled? I've made a quick hack to test USB, but I don't know where is the proper place to do it.

**Comment by ydamigos** [comment]
>Well, it works after enabling HSI48 VREFINT buffer in SYSCFG.
>However, there is a problem: where should SYSCFG itself be enabled? I've made a quick hack to test USB, but I don't know where is the proper place to do it.

@tagunil Thanks, I will check it.

**Comment by ydamigos** [comment]
@tagunil SYSCFG is enabled in soc-gpio.c file for stm32l0 series. I added a check which will raise an error if it is disabled.

**Comment by tagunil** [comment]
@ydamigos It is enabled here only when some EXTI interrupt is configured. I think we should enable it unconditionally somewhere.

**Comment by tagunil** [comment]
@ydamigos Looks good to me. It seems F0 needs this too, because of pin remapping for small packages.

**Comment by ydamigos** [comment]
@tagunil I applied the changes. Could you test it again? I would do the same for F0 family on a separate PR. My only problem is that I don't have the hw to test it.

**Comment by tagunil** [comment]
@ydamigos How about enabling SYSGFG by direct LL calls as in F4/L4 instead of calling the same driver through generic clock control code?

**Comment by ydamigos** [comment]
>How about enabling SYSGFG by direct LL calls as in F4/L4 instead of calling the same driver through generic clock control code?

I moved the code as it was. Where do we use in Zephyr's code LL calls to enable SYSCFG? I cannot find them.

**Comment by tagunil** [comment]
> I moved the code as it was. Where do we use in Zephyr's code LL calls to enable SYSCFG? I cannot find them.
Not SYSCFG, but PWR is enabled by LL calls in stm32f4x_ll_clock.c and stm32l4x_ll_clock.c. I think SYSCFG could be enabled the similar way.

**Comment by ydamigos** [comment]
>Not SYSCFG, but PWR is enabled by LL calls in stm32f4x_ll_clock.c (I was wrong about L4). I think SYSCFG could be enabled the similar way.

I made the change.

**Comment by tagunil** [comment]
> I made the change.

There is a typo in function name: LL_APB2_GPR1_EnableClock should be LL_APB2_GRP1_EnableClock

**Comment by tagunil** [comment]
After fixing the typo I've tried cdc_acm sample, it works just fine.

**Comment by ydamigos** [comment]
>After fixing the typo I've tried cdc_acm sample, it works just fine.

I fixed the typo. Thanks for testing.

**Review by jfischer-no — COMMENTED** [review]
LGTM

**Review by erwango — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Pull request merged**